test/google_search_data_viewer/keywords/schemas/keyword_upload_test.exs

```diff
@@ -0,0 +1,22 @@
+defmodule GoogleSearchDataViewer.Keywords.Schemas.KeywordUploadTest do
+  use GoogleSearchDataViewer.DataCase, async: true
+
+  alias GoogleSearchDataViewer.Keywords.Schemas.KeywordUpload
+
+  # name+user_id
+
+  describe "changeset/2" do
+    test "given a changeset with keyword name and existing user id, validates" do
```

Comment:
```suggestion
    test "given a changeset with keyword name and existing user id, returns valid changeset" do
    # ... 
    assert keyword_upload_changeset.name == "cat"
    assert keyword_upload_changeset.user_id == 1
```

Reply:
Yes thanks I will change that. Although it's not strictly true regarding existing user id - this won't be validated to exist in the changeset

Reply:
in this case, I think we can factory the user

```elixir
test "given a changeset with keyword name and existing user id, validates" do
      %{id: user_id} = insert(:user)

      keyword_upload_changeset =
        KeywordUpload.changeset(%KeywordUpload{}, %{name: "cat", user_id: user_id})

      assert keyword_upload_changeset.valid? == true
      assert keyword_upload_changeset.changes == %{name: "cat", user_id: user_id}
    end
```

Then we can test the association by another test

```elixir
    test "given invalid user, return invalid changeset" do
      keyword_upload_changeset = KeywordUpload.changeset(%KeywordUpload{}, %{name: "cat", user_id: -1})

      assert {:error, changeset} = Repo.insert(keyword_upload_changeset)

      assert errors_on(changeset) == %{user: ["does not exist"]}
    end
```